client/my-sites/marketing/index.node.js

```diff
@@ -0,0 +1,17 @@
+import { makeLayout } from 'calypso/controller';
+
+export default function ( router ) {
+	// Handle redirects on the server.
+	router( [ '/marketing/do-it-for-me', '/marketing/do-it-for-me/*' ], ( { res } ) => {
+		res.redirect( 301, 'https://wordpress.com/do-it-for-me/' );
+	} );
+	router(
+		[ '/marketing/ultimate-traffic-guide', '/marketing/ultimate-traffic-guide/*' ],
+		( { res } ) => {
+			res.redirect( 301, 'https://wpcourses.com/course/intro-to-search-engine-optimization-seo/' );
+		}
+	);
+
+	// Otherwise, use client-side routing.
+	router( [ '/marketing', '/marketing/*', '/sharing', '/sharing/*' ], makeLayout );
```

Comment:
Curious: how complete is the server-side rendering in Calypso? Is it completely isomorphic, meaning data is also fetched and rendered in the server? In this case, how does using `makeLayout` as the callback there lead to client-side routing taking place?

Reply:
I think number 1 is possible, but in this case I wonder if there is some webpack magic happening? @tyxla may know more

Reply:
There's no webpack magic whatsoever in that area 😉 `makeLayout()` uses `makeLayoutMiddleware()`, and that doesn't do any rendering, it only prepares the layout component in the context, but doesn't render it. Actual server rendering happens a bit later, and only for logged-out users (plus [a few more conditions](https://github.com/Automattic/wp-calypso/blob/4f85ff6e974264b1f7f1133827dcfaf395fbdec9/client/server/render/index.js#L296)), and also requires a primary component to be set to be rendered - we don't set it for those routes on that line (look for `context.primary = ...`). So at that point, the route gets handled by the client-side routing.